### PR TITLE
documentation: improve commit structure

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

This commits adds more documentation on how to contribute better commit
messages.

Signed-off-by: Sébastien Han <seb@redhat.com>

**Checklist:**

- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

[skip ci]

### CODE REVIEW

File: Documentation/development-flow.md [hunk 1]
```diff
@@ -239,9 +239,40 @@ go tool cover -html=coverage.out -o coverage.html
...
+Here is an example of an acceptable commit message:
+
+```text
+component: commit title
+
+This is the commit message, here I'm explaining, what the bug was along with its root cause.
+Then I'm explaining how I fixed it.
+
+Closes: https://github.com/rook/rook/issues/<NUMBER>
+Signed-off-by: First Name Last Name <email address>
```
Comment (galexrt) [inline Documentation/development-flow.md]: Shouldn't there be a new line between the last line and the sign-off?

Comment (leseb) [reply Documentation/development-flow.md]: Hum I think it's done by git automatically.

Comment (galexrt) [reply Documentation/development-flow.md]: Sorry I meant a new line between the closes and sign-off, like this:
```
Closes: ...

Signed-off-by:...
```

Comment (leseb) [reply Documentation/development-flow.md]: Oh I see, I've never done that and I don't think it's really aesthetic :-)

**Pull request merged**